barcode_validation/models/barcode_validation_mixin.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class BarcodeValidation(models.AbstractModel):
+    _name = "barcode.validation.mixin"
+    _description = "Barcode Validation"
+
+    def _get_validation_barcode_settings(self):
+        return {
```

Comment:
IMHO, I think having these individual behaviours is not really useful. As you want to display a better error message than the Odoo one (for all barcodes).

IMHO, if you want to introduce other barcode types validation, why not overriding the barcode() function to validate it (and return a ValueError too) ?